datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/BloomCoarseGrainDataMap.java

```diff
@@ -331,8 +332,18 @@ private BloomQueryModel buildQueryModelInternal(CarbonColumn carbonColumn,
       // for dictionary/date columns, convert the surrogate key to bytes
       internalFilterValue = CarbonUtil.getValueAsBytes(DataTypes.INT, convertedValue);
     } else {
-      // for non dictionary dimensions, is already bytes,
-      internalFilterValue = (byte[]) convertedValue;
+      // for non dictionary dimensions, numeric columns will be of original data,
+      // so convert the data to bytes
+      if (DataTypeUtil.isPrimitiveColumn(carbonColumn.getDataType())) {
+        if (convertedValue == null) {
+          convertedValue = DataConvertUtil.getNullValueForMeasure(carbonColumn.getDataType(),
+              carbonColumn.getColumnSchema().getScale());
+        }
+        internalFilterValue =
+            CarbonUtil.getValueAsBytes(carbonColumn.getDataType(), convertedValue);
```

Comment:
Result of `getValueAsBytes`  conflicts with existing bloom index data which will affect query result.
For measure in  'NoDict',  original implementation used `NonDictionaryFieldConverterImpl` to convert value, but now it uses `MeasureFieldConverterImpl` to convert value

Reply:
The above problem is similar to your second commit for minmax

Reply:
> I ran a test on table with bloom datamap created before applying this PR, and query it after this PR merged, but the answer is not correct. Can you check it?
> 
> Procedure to reproduce:
> 
> * switch master code before this PR merged
> * create table with no-dict measure column (set the measure column as sort column)
> * create bloom datamap on the measure column
> * load some data into table
> * query on the measure column, get a result
> * switch to code after this PR merged
> * do the same query and compare the result

I will check this issue and update asap